wrangler-transform/src/main/java/io/cdap/wrangler/Wrangler.java

```diff
@@ -498,6 +500,16 @@ public static class Config extends PluginConfig {
     @Macro
     private String precondition;
 
+    @Name(NAME_PRECONDITION_SQL)
+    @Description("Precondition expression specifying filtering before applying directives (true to filter)")
+    @Macro
+    private String preconditionSql;
+
+    @Name(NAME_PRECONDITION_TOGGLE)
+    @Description("Precondition toggle to toggle between JEXL and SQL")
+    @Macro
+    private String preconditionToggle;
+
     @Name(NAME_DIRECTIVES)
```

Comment:
We also need to update the validation rules to account for the missing Precondition field, and ensure all fields are set correctly.

Reply:
You can check the value of `precondition` based on the value of toggle instead of setting it as `required`.

Reply:
So, in that case can't we use just one single textbox, instead of two?

Reply:
Simple answer is that a Jexl expression and a SQL Expression are incompatible, so it makes sense that these 2 settings are completely independent from each other.